pkg/windows/create_build_env.ps1

```diff
@@ -51,49 +58,74 @@ if (!($7zip)) {
 if (Test-Path .\Salt-Dev) {
     Remove-Item -Recurse -Force Salt-Dev
 }
-md Salt-Dev
-pushd Salt-Dev
+mkdir Salt-Dev
+Push-Location Salt-Dev
 git clone https://github.com/saltstack/salt
-cd salt\pkg\windows
+Set-Location salt\pkg\windows
 git checkout v2016.11.3
 powershell -file build_env.ps1 -Silent
-popd
-pushd Salt-Dev\salt
+Pop-Location
+Push-Location Salt-Dev\salt
 reloadEnv
 python -m pip install --upgrade pip
 pip install -e .
-popd
+Pop-Location
 
 # Install hubble and dependency
 if (Test-Path .\hubble) {
     Remove-Item -Recurse -Force hubble
 }
-git clone https://github.com/hubblestack/hubble
-pushd hubble\pkg\scripts
-$lines = Get-Content pyinstaller-requirements.txt | Where {$_ -notmatch '^\s+$'} 
+#Checks to see if any paramaters were given for both $repo and $branch.
+if ($default){
+    $repo = "https://github.com/hubblestack/hubble"
+    $branch = "master"
+}
+#If no default was specified and no paramaters were given in the script, it prompts for a repo and branch
+if ($repo -notlike "https*"){
+    $repo = Read-Host "Enter a Repository (full URL only)"
+    $branch = Read-Host "Enter a Branch"
+}
+git clone $repo
+Push-Location hubble\pkg\scripts
+if ($branch -like "\W.+") {
+    git checkout $branch
+}
+$lines = Get-Content pyinstaller-requirements.txt | Where-Object {$_ -notmatch '^\s+$'} 
 foreach ($line in $lines) {
     $line = $line -replace "#.+$",""
     if ($line -notlike '*pyinotify*' -and $line -notlike '*salt-ssh*') { #pyinotify and salt-ssh are for linux only
         pip install $line
     }
 }
-popd
+Pop-Location
 
 # Download PortableGit.  Requirement for Hubble
-$path = (Get-Location).Path
-If (Test-Path "C:\Program Files (x86)") {
-    Invoke-WebRequest -Uri https://github.com/git-for-windows/git/releases/download/v2.12.2.windows.2/PortableGit-2.12.2.4-64-bit.7z.exe -OutFile .\PortableGit.7z.exe
+$port_git = choco list --localonly | Where-Object {$_ -like "git.portable *"}
+if (!($port_git)) {
+    choco install git.portable -y
+    reloadEnv
 } else {
-    Invoke-WebRequest -Uri https://github.com/git-for-windows/git/releases/download/v2.12.2.windows.2/PortableGit-2.12.2.4-32-bit.7z.exe -OutFile .\PortableGit.7z.exe
+    if (($port_git -replace "git.portable ","") -lt $portgitVer) {
+        choco upgrade git.portable -y
+    }
+}
+#Moves Portable Git into the correct location to work with hubble. 
+set-location C:\Temp
+Import-Module "$env:ChocolateyInstall\helpers\chocolateyInstaller.psm1" -Force;
+$ChocoTools = Get-ToolsLocation
+if(test-path .\hubble\PortableGit\){
+    Continue
+}
+else{
+    mkdir C:\Temp\hubble\PortableGit\
 }
-7z x PortableGit.7z.exe -o"$path\hubble\PortableGit" -y
-& "$path\hubble\PortableGit\post-install.bat"
+Move-Item -Path $ChocoTools\git\ -Destination C:\Temp\hubble\PortableGit\ 
 
 # Install osquery for executible
 if (!(Test-path C:\ProgramData\osquery)) {
 	choco install osquery -y
 } else {
-	choco update osquery -y
+	choco upgrade osquery -y
```

Comment:
if update is being deprecated.  please change update to upgrade everywhere we are updating the version of a package with chocolatey